### PR TITLE
Feature/DCMAW-7374: initial repo setup

### PR DESCRIPTION
**Pull request by alex-bradbury**
Initial setup with existing authentication code

- Added Git hooks
- Added Github flows
- AppAuth login wrapper code
- maven-publish gradle setup
- related build configs

### CODE REVIEW

File: .githooks/post-checkout
Comment (chrisdondig) [inline .githooks/post-checkout:1]: Are you using git lfs for this module? If not, then you don't need this githook

File: .githooks/post-commit
Comment (chrisdondig) [inline .githooks/post-commit:1]: Are you using git lfs for this module? If not, then you don't need this githook

File: .githooks/post-merge
Comment (chrisdondig) [inline .githooks/post-merge:1]: Are you using git lfs for this module? If not, then you don't need this githook

File: .githooks/post-checkout
Comment (alex-bradbury) [reply .githooks/post-checkout:1]: I am not, thank you!

File: .github/conventional_commit_template.txt
Comment (chrisdondig) [inline .github/conventional_commit_template.txt:1]: This should probably be added to the github repo template: https://github.com/govuk-one-login/mobile-android-one-login-template

File: .github/workflows/on_pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+      - name: Run gradle instrumentation tests
+        uses: ./config/actions/gradle-connected-test
+        with:
+          architecture: 'arm64-v8a'
+          device-profile: 'Nexus 6'
+        if: steps.detect-arch.outputs.arch-in-use == 'arm64'
+
+      - name: Run gradle instrumentation tests
+        uses: ./config/actions/gradle-connected-test
```
Comment (chrisdondig) [inline .github/workflows/on_pull_request.yml:65]: NB: The architecture on the gitrunner is currently x86_64 so this version of gradle-connected-test will always run with the default emulator

File: app/src/test/java/uk/gov/android/authentication/ExampleUnitTest.kt
Comment (chrisdondig) [inline app/src/test/java/uk/gov/android/authentication/ExampleUnitTest.kt:1]: Is this still needed?

File: buildLogic/plugins/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+plugins {
+    `kotlin-dsl`
+}
+
+val kotlinVersion by rootProject.extra("1.9.0")
+val sonarqubeVersion by rootProject.extra("4.3.0.3225")
+
+dependencies {
+    listOf(
```
Comment (chrisdondig) [inline buildLogic/plugins/build.gradle.kts:9]: Could these dependencies all come from the version catalog file?

File: config/actions/gradle-connected-test/action.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    required: false
+    default: 'en-GB'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Run gradle connected test
+      uses: reactivecircus/android-emulator-runner@v2
+      with:
+        api-level: 30
```
Comment (chrisdondig) [inline config/actions/gradle-connected-test/action.yml:24]: Shouldn't this be 29?

**Review by chrisdondig — APPROVED** [review]

File: app/src/test/java/uk/gov/android/authentication/ExampleUnitTest.kt
Comment (alex-bradbury) [reply app/src/test/java/uk/gov/android/authentication/ExampleUnitTest.kt:1]: sadly yes for jacoco

File: buildLogic/plugins/build.gradle.kts (hunk 2 above)
Comment (alex-bradbury) [reply buildLogic/plugins/build.gradle.kts:9]: possibly yes, will havea look

Comment (alex-bradbury) [reply buildLogic/plugins/build.gradle.kts:9]: these can't access project version catalog for some reason :(

**Pull request merged**